src/openforms/contrib/zgw/service.py

```diff
@@ -52,22 +55,22 @@ def create_zaak(
 
 def default_get_drc() -> Service:
     config = ZgwConfig.get_solo()
-    return config.drc_service
+    zgw_config = config.default_zgw_api_group
+    return zgw_config.drc_service
 
 
-def partial_update_zaak(zaak_url: str, data: dict) -> dict:
-    config = ZgwConfig.get_solo()
-    client = config.zrc_service.build_client()
+def partial_update_zaak(zaak_url: str, data: dict, zrc_service: Service) -> dict:
+    client = zrc_service.build_client()
     zaak = client.partial_update("zaak", data, url=zaak_url)
     return zaak
 
 
-def set_zaak_payment(zaak_url: str, partial: bool = False) -> dict:
+def set_zaak_payment(zaak_url: str, service: Service, partial: bool = False) -> dict:
```

Comment:
see above remarks about passing `client` instead of `service` and order of arguments